src/main.js

```diff
@@ -23,40 +23,51 @@ import Vue from "vue";
 
 import Nextcloud from "./mixins/Nextcloud";
 import Recommendations from "./components/Recommendations";
+import Axios from "nextcloud-axios";
+import { generateUrl } from "nextcloud-server/dist/router";
 
 Vue.mixin(Nextcloud);
 
+
 OC.Plugins.register('OCA.Files.FileList', {
 
 	attach: fileList => {
-		if (fileList.id !== 'files') {
-			return;
-		}
-		const el = document.createElement('div');
-		const controls = document.getElementById('controls');
-		controls.parentNode.insertBefore(el, controls.nextSibling);
-		el.id = 'files-recommendation-wrapper';
-
-		const View = Vue.extend(Recommendations);
-		const vm = new View({
-			propsData: {}
-		}).$mount(el);
-
-		fileList.$el.on('changeDirectory', data => {
-			if (data.dir.toString() === '/') {
-				vm.show();
-			} else {
-				vm.hide();
+		Axios.get(generateUrl('/apps/recommendations/settings')).then((res) => {
```

Comment:
:-1: I don't like this as it causes two requests.

Why don't you just return an empty recommendations array when they are disabled? Then you don't need any changes here as an empty list isn't rendered anyway.

Reply:
Can I merge in [this branch](https://github.com/gary-kim/recommendations/tree/add-toggle-controller) then? It changes the controller rather then the client side.

Reply:
Yes, that looks better!